src/core/mesh/qgsmeshlayerutils.cpp

```diff
@@ -659,6 +659,11 @@ QVector<QVector3D> QgsMeshLayerUtils::calculateNormals( const QgsTriangularMesh
       const int index1( face.at( ( i + 1 ) % 3 ) );
       const int index2( face.at( ( i + 2 ) % 3 ) );
 
+      if ( std::isnan( verticalMagnitude[index] ) ||
```

Comment:
isnan or isfinite? can verticalMagnitude[i]  be infinite?

Reply:
isnan, infinite is not considered/handle by mesh layer dataset values.
isnan is the no data value for mesh layers.